==13279== Memcheck, a memory error detector
==13279== Copyright (C) 2002-2017, and GNU GPL'd, by Julian Seward et al.
==13279== Using Valgrind-3.13.0 and LibVEX; rerun with -h for copyright info
==13279== Command: /home/wasmer/repositories/wasmer/wasmer/target/release/examples/engine-native
==13279== 
--13279-- WARNING: unhandled amd64-linux syscall: 332
--13279-- You may be able to write your own handler.
--13279-- Read the file README_MISSING_SYSCALL_OR_IOCTL.
--13279-- Nevertheless we consider this a bug.  Please report
--13279-- it at http://valgrind.org/support/bug_reports.html.
==13279== Conditional jump or move depends on uninitialised value(s)
==13279==    at 0x6B5F4D: wasmparser::validator::Validator::end (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/engine-native)
==13279==    by 0x6ABA4C: wasmparser::validator::Validator::payload (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/engine-native)
==13279==    by 0x6AACC1: wasmparser::validator::Validator::validate_all (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/engine-native)
==13279==    by 0x2C428C: wasmer_compiler::compiler::Compiler::validate_module (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/engine-native)
==13279==    by 0x26DA58: <wasmer_engine_native::engine::NativeEngine as wasmer_engine::engine::Engine>::validate (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/engine-native)
==13279==    by 0x5A6224: wasmer::module::Module::from_binary (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/engine-native)
==13279==    by 0x2664D1: wasmer::module::Module::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/engine-native)
==13279==    by 0x268A1F: engine_native::main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/engine-native)
==13279==    by 0x266A22: std::rt::lang_start::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/engine-native)
==13279==    by 0x758082: {{closure}} (rt.rs:52)
==13279==    by 0x758082: do_call<closure-0,i32> (panicking.rs:348)
==13279==    by 0x758082: try<i32,closure-0> (panicking.rs:325)
==13279==    by 0x758082: catch_unwind<closure-0,i32> (panic.rs:394)
==13279==    by 0x758082: std::rt::lang_start_internal (rt.rs:51)
==13279==    by 0x269891: main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/engine-native)
==13279== 
==13279== 
==13279== HEAP SUMMARY:
==13279==     in use at exit: 30,888 bytes in 101 blocks
==13279==   total heap usage: 728 allocs, 627 frees, 124,935 bytes allocated
==13279== 
==13279== 1,152 bytes in 4 blocks are possibly lost in loss record 35 of 43
==13279==    at 0x4C33B25: calloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13279==    by 0x4013666: allocate_dtv (dl-tls.c:286)
==13279==    by 0x4013666: _dl_allocate_tls (dl-tls.c:530)
==13279==    by 0x5252227: allocate_stack (allocatestack.c:627)
==13279==    by 0x5252227: pthread_create@@GLIBC_2.2.5 (pthread_create.c:644)
==13279==    by 0x7666FC: std::sys::unix::thread::Thread::new (thread.rs:66)
==13279==    by 0x307A76: std::thread::Builder::spawn (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/engine-native)
==13279==    by 0x30ADF6: <rayon_core::registry::DefaultSpawn as rayon_core::registry::ThreadSpawn>::spawn (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/engine-native)
==13279==    by 0x30B9AF: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/engine-native)
==13279==    by 0x30E9A6: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/engine-native)
==13279==    by 0x751729: std::sync::once::Once::call_inner (once.rs:416)
==13279==    by 0x30AF70: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/engine-native)
==13279==    by 0x30DD6B: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/engine-native)
==13279==    by 0x2B6F51: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/engine-native)
==13279== 
==13279== 1,432 (416 direct, 1,016 indirect) bytes in 1 blocks are definitely lost in loss record 36 of 43
==13279==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13279==    by 0x6D521D: wasmer_vm::instance::InstanceHandle::allocate_instance (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/engine-native)
==13279==    by 0x276ECD: wasmer_engine::artifact::Artifact::instantiate (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/engine-native)
==13279==    by 0x5A65DD: wasmer::module::Module::instantiate (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/engine-native)
==13279==    by 0x5ABDD2: wasmer::instance::Instance::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/engine-native)
==13279==    by 0x268B7E: engine_native::main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/engine-native)
==13279==    by 0x266A22: std::rt::lang_start::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/engine-native)
==13279==    by 0x758082: {{closure}} (rt.rs:52)
==13279==    by 0x758082: do_call<closure-0,i32> (panicking.rs:348)
==13279==    by 0x758082: try<i32,closure-0> (panicking.rs:325)
==13279==    by 0x758082: catch_unwind<closure-0,i32> (panic.rs:394)
==13279==    by 0x758082: std::rt::lang_start_internal (rt.rs:51)
==13279==    by 0x269891: main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/engine-native)
==13279== 
==13279== LEAK SUMMARY:
==13279==    definitely lost: 416 bytes in 1 blocks
==13279==    indirectly lost: 1,016 bytes in 12 blocks
==13279==      possibly lost: 1,152 bytes in 4 blocks
==13279==    still reachable: 28,304 bytes in 84 blocks
==13279==         suppressed: 0 bytes in 0 blocks
==13279== Reachable blocks (those to which a pointer was found) are not shown.
==13279== To see them, rerun with: --leak-check=full --show-leak-kinds=all
==13279== 
==13279== For counts of detected and suppressed errors, rerun with: -v
==13279== Use --track-origins=yes to see where uninitialised values come from
==13279== ERROR SUMMARY: 3 errors from 3 contexts (suppressed: 0 from 0)
